test/framework/actions.ts

```diff
@@ -0,0 +1,12 @@
+import * as request from 'supertest';
+
+const API_HTTP_ADDRESS = 'http://localhost:3000';
+
+/**
+ * Makes a HTTP GET request.
+ * @param input Specifies the route and the base URL that should be used to make
+ *        the HTTP GET request.
+ */
+export async function apiGetRequestAsync(input: { route: string; baseURL?: string }): Promise<request.Response> {
+    return request(input.baseURL || API_HTTP_ADDRESS).get(input.route);
```

Comment:
How about lifting the `request` options interface?

Something like this:
```
function requestAsync(requestOpts: RequestOpts): Promise<request.Response> {
    return request(requestOpts);
}
```

I believe `request` options are defined as `CoreOptions` in `@types/request`. 

Essentially this function would be a thin wrapper around the request library of our choice, and we wouldn't have to write a whole bunch of boilerplate around handling different types of requests. Would also make it easier to switch out for a different request library down the road (`request` is deprecated as of February 2020!)

Another nice feature might be to persist the `request` module in the TestManager and allow the user to set defaults by calling `request.defaults(options)`. This is more of a nice-to-have, feel free to pass for now.